codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java

```diff
@@ -0,0 +1,145 @@
+import software.amazon.smithy.aws.go.codegen.AwsSignatureVersion4;
+import software.amazon.smithy.aws.go.codegen.AwsSignatureVersion4aUtils;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.aws.traits.auth.SigV4ATrait;
+import software.amazon.smithy.aws.traits.auth.SigV4Trait;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDelegator;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.ConfigField;
+import software.amazon.smithy.go.codegen.integration.ConfigFieldResolver;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.traits.AuthTrait;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SetUtils;
+
+/**
+ * This integration configures the CloudFront Key Value Store client for Signature Version 4a
+ */
+public class CloudFrontKVSSigV4a implements GoIntegration {
+    /**
+     * Return true if service is CFKVS.
+     *
+     * @param model   is the generation model.
+     * @param service is the service shape being audited.
+     */
+    private static boolean isCFKVSService(Model model, ServiceShape service) {
+        final String sdkId = service.expectTrait(ServiceTrait.class).getSdkId();
+        final String serviceId = sdkId.replace("-", "").replace(" ", "").toLowerCase();
+        return serviceId.equalsIgnoreCase("cloudfrontkeyvaluestore");
+    }
+
+    private final List<RuntimeClientPlugin> runtimeClientPlugins = new ArrayList<>();
+
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return runtimeClientPlugins;
+    }
+
+    @Override
+    public Model preprocessModel(Model model, GoSettings settings) {
+        ServiceShape service = settings.getService(model);
+        if (!isCFKVSService(model, service)) {
+            return model;
+        }
+
+        if (settings.getService(model).hasTrait(SigV4ATrait.class)) {
+            return model;
+        }
+
+        var v4a = SigV4ATrait.builder()
+                .name(service.expectTrait(SigV4Trait.class).getName())
+                .build();
+
+        return model.toBuilder()
+                .addShape(
+                        service.toBuilder()
+                                .addTrait(v4a)
+                                .addTrait(new AuthTrait(SetUtils.of(SigV4ATrait.ID, SigV4Trait.ID)))
```

Comment:
```suggestion
                                .addTrait(v4a)
                                // FUTURE: https://github.com/aws/smithy-go/issues/493
                                // we are keeping sigv4 at the end of this list (it will never be selected) as a stopgap to drive codegen of payload checksum routines
                                .addTrait(new AuthTrait(SetUtils.of(SigV4ATrait.ID, SigV4Trait.ID)))
```